src/widget/wlibrary.cpp

```diff
@@ -39,7 +39,12 @@ void WLibrary::switchToView(const QString& name) {
         if (widget != NULL && currentWidget() != widget) {
             //qDebug() << "WLibrary::setCurrentWidget" << name;
             setCurrentWidget(widget);
-            dynamic_cast<LibraryView*>(widget)->onShow();
+            LibraryView * lview = dynamic_cast<LibraryView*>(widget);
+              if (lview == NULL) {
```

Comment:
if you like you could also fix the double lookup using contains() and [] some lines above
Just use a single QMap::value(const Key & key, const T & defaultValue = T()) call.